Ti comm dev

ihagedo opened the pull request:
## What is a quick description of the change?
Merging finished serial comm modules into the master.

## Is this fixing an issue?
#21 

## Were any issues created as a result of this change?
None.

## Are there more details that are relevant?
Fixed library dependency issues and created a better modularized code for debugging.

Future fixes include packet depreciation over time for echo call-back functionality.

## Check lists (check x in [ ] of list items)
- [ ] Test written/updating
- [ ] Tests passing
- [ ] Coding style (indentation, etc)

## Any additional comments?

ihagedo committed:
modified sizes of objects

zghera committed:
Modify makefile and SERIAL_TEST to try to find issue with makefile.

ihagedo committed:
bag test

ihagedo committed:
deleted outdated file

ihagedo committed:
added bagfile test

ihagedo committed:
Merge branch 'isaac_dev' of https://github.com/Autonomous-Motorsports-Purdue/AMP-v1 into isaac_dev

ihagedo committed:
added bagfile test data

ihagedo committed:
Merge branch 'isaac_dev' of https://github.com/Autonomous-Motorsports-Purdue/AMP-v1 into isaac_dev

ihagedo committed:
serial comm

Commit:
Merge branch 'ti_comm_dev' of https://github.com/Autonomous-Motorsports-Purdue/AMP-v1 into ti_comm_dev

Reset origin

Commit:
Serial Communiction files updated. ECU_TEST works well with debug option. SERIAL_TEST is for scaled up testing.

ihagedo committed:
Update .gitattributes

clean attributes

ihagedo committed:
cleaning lfs

ihagedo committed:
added libserialport submodule

ihagedo committed:
added readme

ihagedo committed:
Update README.md

ihagedo committed:
Merge pull request #14 from Autonomous-Motorsports-Purdue/master

updating isaac-dev

ihagedo committed:
fixed structural issues for packet

ihagedo committed:
Merge branch 'isaac_dev' into ti_comm_dev

ihagedo committed:
Update amp_serial_jetson_ECU_TEST.cpp

cleaned up ECU file

ihagedo committed:
Delete amp_serial_jetson_ECU_TEST.o

removed object file

ihagedo committed:
Delete amp_serial_jetson

remove object file

ihagedo committed:
Delete amp_serial_jetson_SERIAL_TEST.o

remove object file

ihagedo committed:
Update amp_serial_jetson.h

uncommented declarations

zghera commented on .gitattributes:
```diff
@@ -0,0 +1 @@
+
```
What is the purpose of this empty file?

zghera commented on README.md:
````diff
@@ -11,8 +11,14 @@ rosdep install --from-paths src --ignore-src -r -y
 source devel/setup.bash
 ```
+#### LIBSERIALPORT
+At the top of each branch your want to build the library for you should run the following command and then refer to the readme in src/ti_comm/src
+```
+git submodule update --init --recursive
+```
````
```suggestion 
The serial communication node relies on a serial communication library that needs to be built each time you check out a new branch (including master when initially cloning the repo). To properly build the library, run the following commands from the top of the repo directory:
1. `git submodule update --init --recursive`
2. `cd src/ti_comm/src/libserialport`
3. `./autogen.sh`
4. `./configure`
5. `make`
6. `sudo make install`
7. If you don't see something like `libserialport` in your PATH environment variable (`echo $PATH), manually add it to your Path.
```

zghera commented on src/ti_comm/src/README.md:
```diff
@@ -0,0 +1,10 @@
+How to Build Libserialport
```
Remove this file as a whole if you agree with the suggestion to move its contents t the top-level README

zghera reviewed (commented):
After trying the install, I ran across a couple of things that we should probably chat offline about before I do any more reviews. While the submodules init seemed to work fine, the `autogen.sh` did not. I also ran catkinmake to see what would happen and ended up getting some import errors for libserialport. I looked around and also saw that the CMakeLists.txt config does not look updated. 

Updated the instructions after our conversation. While the install of libserialport seemed to work fine, there are still two issues that we know of
1. When changing the references of `libserialport` in CMakiLists.txt to the new folder, there are now errors that result when catkinmake-ing due to warnings in some of the included files.
2. The ti_comm node does not show up in the node graph.

Isaac and Zach are currently looking into issues 1 and 2, respectively.

zghera commented:
> After trying the install, I ran across a couple of things that we should probably chat offline about before I do any more reviews. While the submodules init seemed to work fine, the `autogen.sh` did not. I also ran catkinmake to see what would happen and ended up getting some import errors for libserialport. I looked around and also saw that the CMakeLists.txt config does not look updated.
> 
> Updated the instructions after our conversation. While the install of libserialport seemed to work fine, there are still two issues that we know of
> 
> 1. When changing the references of `libserialport` in CMakiLists.txt to the new folder, there are now errors that result when catkinmake-ing due to warnings in some of the included files.
> 2. The ti_comm node does not show up in the node graph.
> 
> Isaac and Zach are currently looking into issues 1 and 2, respectively.

Update on issue 2 here: When reverting to the old CMakeLists.txt file (build the liberialport.0.1.0 lib not the new one), the ti_comm ROS node did not show up because of a memory error when running the node (see below). This is possible due to the imports in `amp_serial_jetson.cpp` not using the correct libserialport imports or something similar. So it is evident that issue 1 listed above (warnings when changing CMakeLists.txt to correct libserialport files) must be solved before anything else.

Commit:
added number parameters

ihagedo committed:
seperated library files

zghera reviewed (changes requested):
Linked Issue #21. Will revisit reviewing this PR once we work on it later this week @ihagedo.

ihagedo committed:
integrated new library into build

Commit:
Merge branch 'ti_comm_dev' of https://github.com/Autonomous-Motorsports-Purdue/AMP-v1 into ti_comm_dev

ihagedo committed:
ecu test working properly

bitmaks committed:
Gitignore vim swap files

bitmaks committed:
Add read command from file functionality and delay flag to debugger

bitmaks committed:
Improve help description and add exit after help menu

bitmaks committed:
Simplify speed and angle read function

Co-authored-by: Zach Ghera <zpghera00@gmail.com>

bitmaks committed:
Refactor to be more readable

bitmaks committed:
Merge pull request #25 from aaaakshat/ti_comm_dev

Read commands from file + specify delay at run (Ti_comm_dev)

Commit:
Merge branch 'ti_comm_dev' of https://github.com/Autonomous-Motorsports-Purdue/AMP-v1 into ti_comm_dev

Commit:
restructured ti_comm dir

Commit:
Merge branch 'master' of https://github.com/Autonomous-Motorsports-Purdue/AMP-v1 into ti_comm_dev

Commit:
Added includes for SERIAL_TEST

Commit:
segrated old and new lib, resolved new lib build issues

ihagedo committed:
restructuring ti_comm/src

bitmaks committed:
Add libserialport patch for linux configs

bitmaks committed:
Fix patch whitespacing issue

bitmaks committed:
Add build script

zghera committed:
Modified build.sh to create libserialport obj files.

bitmaks committed:
Recognize if patch has been applied in build.sh

bitmaks committed:
FIx scriptDir error on linux

bitmaks committed:
Fix typo

zghera committed:
Change shell to bash so source works.

Also added some additional comments.

bitmaks committed:
Don't fetch libserialport if patch has been applied

bitmaks committed:
Gitignore MacOS trash and debug files

ihagedo committed:
fixed ECU

ihagedo committed:
Merge branch 'ti_comm_dev' of github.com:Autonomous-Motorsports-Purdue/AMP_ASSv1 into ti_comm_dev

ihagedo committed:
added micro sleep

bitmaks committed:
Fix serial pkt and control pkt memcpy issue

This fixes the issue with memcpy call on control packet by changin control packet v_speed and v_angle from int to char in amp_serial_jetson.h. Previously this caused an issue where serial packet looked in msg[0] and msg[1] bytes to get the values of speed and angle, but the control packet was an 8 byte struct of 2 integers which caused msg[0] and msg[1] to be the values of the first int in the struct.

bitmaks committed:
Make control pkt use unsigned not signed char

bitmaks committed:
Add debug messages to serial and ecu test

bitmaks committed:
serial test work

bitmaks committed:
Update amp_serial_jetson (please work)

Commit:
Debugging 0 serial packets. ti_comm node is crapping out now.
